### PR TITLE
WebDriver+Friends: Support running headless WebDriver sessions

### PR DESCRIPTION
**Pull request by trflynn89**
Overall this:
* Moves functionality needed by both WebContent and headless-browser for WebDriver out of `PageHost` and into `Page`
* Adds an extension capability to let clients choose on a per-session basis if the remote end should be headless
* Runs headless-browser instead of Browser when that capability is set

There's probably a lot of WebDriver endpoints that don't work, because most of headless-browser's `Web::PageClient` is a stub. But navigating / screenshots work!

Ladybird counterpart: https://github.com/SerenityOS/ladybird/pull/135

[Screencast from 2022-11-22 08-13-00.webm](https://user-images.githubusercontent.com/5600524/203323744-6d4716b2-b50d-41f4-962a-3d28ab070894.webm)

### CODE REVIEW

File: Userland/Services/WebContent/WebDriverConnection.h [hunk 1]
```diff
@@ -106,7 +106,7 @@ class WebDriverConnection final
     void delete_cookies(Optional<StringView> const& name = {});
 
     ConnectionFromClient& m_web_content_client;
-    PageHost& m_page_host;
+    Web::PageClient& m_page_host;
```
Comment (linusg) [inline Userland/Services/WebContent/WebDriverConnection.h:108]: Should this member be renamed now?

Comment (trflynn89) [reply Userland/Services/WebContent/WebDriverConnection.h]: :+1:  sure

File: Userland/Services/WebContent/WebDriverConnection.cpp [hunk 2]
```diff
@@ -251,7 +250,7 @@ static ErrorOr<PropertyType, Web::WebDriver::Error> get_property(JsonValue const
     }
 }
 
-ErrorOr<NonnullRefPtr<WebDriverConnection>> WebDriverConnection::connect(ConnectionFromClient& web_content_client, PageHost& page_host, String const& webdriver_ipc_path)
+ErrorOr<NonnullRefPtr<WebDriverConnection>> WebDriverConnection::connect(ConnectionFromClient& web_content_client, Web::PageClient& page_host, String const& webdriver_ipc_path)
```
Comment (linusg) [inline Userland/Services/WebContent/WebDriverConnection.cpp]: `Web::PageClient& page_host` -> `Web::PageClient& page_client` here as well :)

File: Userland/Services/WebContent/WebDriverConnection.cpp [hunk 3]
```diff
@@ -260,10 +259,10 @@ ErrorOr<NonnullRefPtr<WebDriverConnection>> WebDriverConnection::connect(Connect
     return try_create(move(socket), web_content_client, page_host);
 }
 
-WebDriverConnection::WebDriverConnection(NonnullOwnPtr<Core::Stream::LocalSocket> socket, ConnectionFromClient& web_content_client, PageHost& page_host)
+WebDriverConnection::WebDriverConnection(NonnullOwnPtr<Core::Stream::LocalSocket> socket, ConnectionFromClient& web_content_client, Web::PageClient& page_host)
     : IPC::ConnectionToServer<WebDriverClientEndpoint, WebDriverServerEndpoint>(*this, move(socket))
     , m_web_content_client(web_content_client)
-    , m_page_host(page_host)
+    , m_page_client(page_host)
```
Comment (linusg) [inline Userland/Services/WebContent/WebDriverConnection.cpp:265]: and here

**Pull request merged**